mm2src/coins/utxo/utxo_block_header_storage/wasm/indexeddb_block_header_storage.rs

```diff
@@ -207,44 +207,45 @@ impl BlockHeaderStorageOps for IDBBlockHeadersStorage {
         &self,
         max_bits: u32,
     ) -> Result<Option<BlockHeader>, BlockHeaderStorageError> {
-        let ticker = self.ticker.clone();
+        let ticker = &self.ticker;
         let locked_db = self
             .lock_db()
             .await
-            .map_err(|err| BlockHeaderStorageError::get_err(&ticker, err.to_string()))?;
+            .map_err(|err| BlockHeaderStorageError::get_err(ticker, err.to_string()))?;
         let db_transaction = locked_db
             .get_inner()
             .transaction()
             .await
-            .map_err(|err| BlockHeaderStorageError::get_err(&ticker, err.to_string()))?;
+            .map_err(|err| BlockHeaderStorageError::get_err(ticker, err.to_string()))?;
         let block_headers_db = db_transaction
             .table::<BlockHeaderStorageTable>()
             .await
-            .map_err(|err| BlockHeaderStorageError::table_err(&ticker, err.to_string()))?;
+            .map_err(|err| BlockHeaderStorageError::table_err(ticker, err.to_string()))?;
 
-        let mut cursor = block_headers_db
+        let condition = move |block| {
+            serde_json::from_value::<BlockHeaderStorageTable>(block)
+                .map_to_mm(|err| CursorError::ErrorDeserializingItem(err.to_string()))
+                .map(|header| header.bits != max_bits)
+        };
+        let maybe_next = block_headers_db
             .cursor_builder()
-            .only("ticker", ticker.clone())
-            .map_err(|err| BlockHeaderStorageError::get_err(&ticker, err.to_string()))?
+            .only("ticker", self.ticker.clone())
+            .map_err(|err| BlockHeaderStorageError::get_err(ticker, err.to_string()))?
             // We need to provide any constraint on the `height` property
             // since `ticker_height` consists of both `ticker` and `height` properties.
             .bound("height", BeBigUint::from(0u64), BeBigUint::from(u64::MAX))
             // Cursor returns values from the lowest to highest key indexes.
             // But we need to get the most highest height, so reverse the cursor direction.
             .reverse()
+            .where_(condition)
```

Comment:
What would happen if there are more than one object/record that satisfies the condition? I see that we return the first object, are we sure we won't get `Uncaught Error: closure invoked recursively or after being dropped`. This is part of the reason I wanted `where` and `limit`, so we can get more than one object that satisfies the condition, if we want this in other cases (other than the case here where we want only the first object).

P.S. If we don't get the error when there are more than one object that satisfies the condition, then the inclusion of limit and refactoring of where to return more than one object if they exist can be done in a different PR since this PR purpose is to fix the error only.

Reply:
hmm this seems like another enhancement that's not related to this PR.

This PR only handle case for retrieving the first item that meets a condition(if specified) which solves js exception error: `Uncaught Error: closure invoked recursively or after being` But for sure this PR indeed fixed that and also optimizes calls to `indexeddb`.

I will work on the later issue during next print.